code/modules/projectiles/projectile/special.dm

```diff
@@ -46,7 +46,7 @@
 	name = "freeze beam"
 	icon_state = "ice_2"
 	damage = 0
-	damage_type = BURN
+	damage_type = COLD

```

Comment:
You need to add SetColdDamage() and AdjustColdDamage(), and make health and all other factors count in Cold damage.